clis/tcli/src/bin/dtcli/doc/dtcli_documentation.md

```diff
@@ -0,0 +1,105 @@
+```
+
+### Building The Documentation
+
+The full documentation for DTCLI can be built with the following command:
+
+```shell
+cargo doc -p tcli --bin dtcli --open --no-deps --document-private-items
+```
+
+## How To Use DTCLI
+
+### Prerequisites
+
+DTCLI needs a user group called `nordvpn` to be present on the current machine to work. If you have nordvpn installed, the group probably already exists, but if it doesn't you can add it with `groupadd nordvpn` and add the current user to that group with `usermod -aG nordvpn $USER` and relogin afterwords.
+
+### Usage
+
+The usage of DTCLI is the same as TCLI with one main difference: **DTCLI works in the background so there's no need for an STDIO interface**
+
+So instead of running a TCLI executable and then writing TCLI commands into the STDIO based CLI you write `<path to dtcli executable> <TCLI command>` directly into your terminal **or from a script**.
+
+Here's an example of how to run meshnet using DTCLI:
+
+```shell
+./dtcli mesh on tun10
+```
+
+You can also use DTCLI directly through cargo to build and run using only one command:
+
+```shell
+cargo run -p tcli --bin dtcli -- mesh on tun10
+```
+
+See `example_vpn_enable_linux.sh` and `example_vpn_disable_linux.sh` for examples on how to use DTCLI in a script.
+
+### How To Start DTCLI Daemon
+
+The DTCLI daemon is started when running any command (except for `quit` for obvious reasons), or even no commands so you don't really need to worry about starting it manually.
+
+So if you want to just start the DTCLI daemon, run:
+
+```shell
+./dtcli
+```
+
+### How To Stop DTCLI
+
+Because DTCLI runs as a daemon in the background it needs to be stopped manually using the same `quit` command which is used by TCLI.
+
+So DTCLI can be stopped by running:
+
+```shell
+./dtcli quit
+```
+
+>NOTE: If due to a bug the daemon fails to stop using the `quit` command, you can run `sudo cat /var/run/dtcli_wd/dtcli.pid | sudo xargs kill -9` from the root of the libtelio repo (or wherever the dtcli.pid fie is visible from).
+
+### Feature Flags
+
+Feature flags are specified by putting the -f flag as a json string with no whitespaces:
+
+```shell
+./target/debug/dtcli -f '{"direct":{}}'
+```
+
+>NOTE: Feature flags can only be set when starting the daemon, so if it's already running, you'll need to stop it first.
+
+### Help
+
+Help command works the same way as with TCLI and will not start the DTCLI daemon.
+
+So if you want to find out how to use the device connect command, run:
+
+```shell
+./target/debug/dtcli dev con --help
+```
+
+## Logs
+
+Logs along with other files related to DTCLI can be found at `/var/run/dtcli_wd/` directory.
+
+## Platform Support
+
+- Linux is supported and tested.
+- MacOS is supported, but not tested.
+- Windows is not supported yet.
+
+## Compatibility
+
+All of the examples given here were tested with:
+
+- rustc 1.77.2 (25ef9e3d8 2024-04-09)
+- cargo 1.77.2 (e52e36006 2024-03-26)
+- rustdoc 1.77.2 (25ef9e3d8 2024-04-09)
```

Comment:
👍 👍  for the documentation!

Reply:
You can add `#![deny(missing_docs)]` to maintain future update with relevant documentation

Reply:
Done